core/src/main/java/org/infinispan/encoding/DataConversion.java

```diff
@@ -296,12 +292,12 @@ private static boolean isDefault(DataConversion dataConversion) {
    }
 
    public static void writeTo(ObjectOutput output, DataConversion dataConversion) throws IOException {
-      if (isDefault(dataConversion)) {
-         output.writeByte(1);
-      } else {
-         byte flags = 0;
-         if (dataConversion.isKey) flags = (byte) (flags | 2);
-         output.writeByte(flags);
+      byte flags = 0;
+      boolean isDefault = isDefault(dataConversion);
+      if (isDefault(dataConversion)) flags = 1;
```

Comment:
We could use the newly extracted isDefault flag instead of invoking isDefault(dataConversion) again

Reply:
sure, my bad